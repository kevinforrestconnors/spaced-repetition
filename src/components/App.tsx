import * as React from "react";

type HelloWorldProps = {
  userName: string;
  lang: string;
}

export function App(props: HelloWorldProps): React.ReactElement<HelloWorldProps> {
  return (
    <h1>
      Hi {props.userName} from React! Welcome to {props.lang}!
    </h1>
  );
}