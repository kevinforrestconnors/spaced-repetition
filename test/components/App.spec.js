import { App } from '../../src/components/App';
import React from 'react';
import expect from 'expect';
import renderer from 'react-test-renderer';

describe('App components', () => {
  it('renders props', () => {
    // given
    const userName = 'kevinforrestconnors';
    const userName2 = 'anaanramay';
    const lang = 'typescript';
    const lang2 = 'javascript';
    
    // when
    const tree = renderer.create(
      <App
        userName={userName}
        lang={lang}
      />
    )
    .toJSON();

    const tree2 = renderer.create(
      <App
        userName={userName2}
        lang={lang2}
      />
    )
    .toJSON();
    
    // then
    expect(tree).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
  });
});