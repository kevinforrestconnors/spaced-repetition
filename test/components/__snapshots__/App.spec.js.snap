// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components renders props 1`] = `
<h1>
  Hi 
  kevinforrestconnors
   from React! Welcome to 
  typescript
  !
</h1>
`;

exports[`App components renders props 2`] = `
<h1>
  Hi 
  anaanramay
   from React! Welcome to 
  javascript
  !
</h1>
`;
